pkg/resource/model_package/resource.go

```diff
@@ -0,0 +1,96 @@
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package model_package
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackerrors "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8srt "k8s.io/apimachinery/pkg/runtime"
+
+	svcapitypes "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+)
+
+// Hack to avoid import errors during build...
+var (
+	_ = &ackerrors.MissingNameIdentifier
+)
+
+// resource implements the `aws-controller-k8s/runtime/pkg/types.AWSResource`
+// interface
+type resource struct {
+	// The Kubernetes-native CR representing the resource
+	ko *svcapitypes.ModelPackage
+}
+
+// Identifiers returns an AWSResourceIdentifiers object containing various
+// identifying information, including the AWS account ID that owns the
+// resource, the resource's AWS Resource Name (ARN)
+func (r *resource) Identifiers() acktypes.AWSResourceIdentifiers {
+	return &resourceIdentifiers{r.ko.Status.ACKResourceMetadata}
+}
+
+// IsBeingDeleted returns true if the Kubernetes resource has a non-zero
+// deletion timestemp
+func (r *resource) IsBeingDeleted() bool {
+	return !r.ko.DeletionTimestamp.IsZero()
+}
+
+// RuntimeObject returns the Kubernetes apimachinery/runtime representation of
+// the AWSResource
+func (r *resource) RuntimeObject() k8srt.Object {
+	return r.ko
+}
+
+// MetaObject returns the Kubernetes apimachinery/apis/meta/v1.Object
+// representation of the AWSResource
+func (r *resource) MetaObject() metav1.Object {
+	return r.ko
+}
+
+// RuntimeMetaObject returns an object that implements both the Kubernetes
+// apimachinery/runtime.Object and the Kubernetes
+// apimachinery/apis/meta/v1.Object interfaces
+func (r *resource) RuntimeMetaObject() acktypes.RuntimeMetaObject {
+	return r.ko
+}
+
+// Conditions returns the ACK Conditions collection for the AWSResource
+func (r *resource) Conditions() []*ackv1alpha1.Condition {
+	return r.ko.Status.Conditions
+}
+
+// ReplaceConditions sets the Conditions status field for the resource
+func (r *resource) ReplaceConditions(conditions []*ackv1alpha1.Condition) {
+	r.ko.Status.Conditions = conditions
+}
+
+// SetObjectMeta sets the ObjectMeta field for the resource
+func (r *resource) SetObjectMeta(meta metav1.ObjectMeta) {
+	r.ko.ObjectMeta = meta
+}
+
+// SetIdentifiers sets the Spec or Status field that is referenced as the unique
+// resource identifier
+func (r *resource) SetIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
+	if identifier.NameOrID == "" {
+		return ackerrors.MissingNameIdentifier
+	}
+	r.ko.Spec.ModelPackageName = &identifier.NameOrID
```

Comment:
Discussed on a call with Nicholas but seems to be that ModelPackageName will be going through the describe call from sdkFind and will use arn or name as already added in hookcode. So current code should already work for this case. Am currently testing adopting resource for both cases however.

Reply:
Do you mean you tried specifying ARN in `spec.aws.nameOrID` to adopt for versioned model and it worked? Well, technically it might have worked because of the nature of this API accepting ARN as well as name as input to API but since customer knows versioned model package does not have a name, they will use [spec.aws.arn](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/config/crd/common/bases/services.k8s.aws_adoptedresources.yaml#L49) instead of [spec.aws.nameOrID](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/config/crd/common/bases/services.k8s.aws_adoptedresources.yaml#L53) so I think we should stick to using ARN for both cases. It should work out of the box

Reply:
> Looks like it is able to adopt both resources. Although when deleting the versioned modelpackage through kubectl it removes it from k8s side but still exists on sagemaker.

interesting, I noticed service side check is missing in test. Added a comment. Lets fix this